nano/core_test/ledger.cpp

```diff
@@ -688,6 +690,34 @@ TEST (ledger, delete_rep_weight_of_zero)
 	ASSERT_EQ (0, store->rep_weight.count (txn));
 }
 
+TEST (ledger, rep_cache_min_weight)
+{
+	auto store{ nano::test::make_store () };
+	nano::uint128_t min_weight{ 10 };
+	nano::rep_weights rep_weights{ store->rep_weight, min_weight };
+	auto txn{ store->tx_begin_write () };
+
+	// one below min weight
+	rep_weights.representation_add (txn, 1, 9);
+	ASSERT_EQ (0, rep_weights.size ());
+	ASSERT_EQ (1, store->rep_weight.count (txn));
+
+	// exactly min weight
+	rep_weights.representation_add (txn, 1, 1);
+	ASSERT_EQ (1, rep_weights.size ());
+	ASSERT_EQ (1, store->rep_weight.count (txn));
+
+	// above min weight
+	rep_weights.representation_add (txn, 1, 1);
+	ASSERT_EQ (1, rep_weights.size ());
+	ASSERT_EQ (1, store->rep_weight.count (txn));
+
+	// fall blow min weight
+	rep_weights.representation_add (txn, 1, std::numeric_limits<nano::uint128_t>::max () - 5);
```

Comment:
I think the "0 - x" style is more clear since it's exactly like subtracting x.

Reply:
Fixed it